### ISSUE TITLE
The browser crashes if a selection is perform on a long article from wikipedia

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla-services/pageshot#1535]
[Affected versions]:
- Nightly 51.0a1
- Mozilla Page Shot 0.1.201609141513

[Affected Platforms]:
- All Windows
- All Mac OS

[Notes]:
- Attached the crash report: https://goo.gl/AVRSFF.

[Prerequisites]:
- Latest Page Shot add-on (v0.1.201609141513) is installed on a clean profile.

[Steps to reproduce]:
1  Open the browser with the profile from prerequisite and go to https://en.wikipedia.org/wiki/United_States.
2. Click on the "Page Shot" button from browser toolbar and perform a selection.
3. Observe the browser behavior.

[Expected result]:
- The selection functionality overlay is displayed and a selection can be performed.

[Actual result]:
- The browser crashes after pressing the Page Shot icon from the toolbar. 

[Regression]:
- This issue is not reproducible on the latest Aurora (50.0a2), so I have managed to perform a regression. This are the results:

- Last good revision: ca7974ddccc11ec992834b8563d381d16b56ee42
- First bad revision: 8cc4907776c85b5f57c1778bdd5b2a4ad8eb8055
- Pushlog: https://goo.gl/6SenVv.

I suspect the following bug has the changes which introduced the regression: [1295097.](https://bugzilla.mozilla.org/show_bug.cgi?id=1295097)

**Comment by ianb** [comment]
@SoftVision-CosminMuntean : can you CC me on that bug (1295097)?

**Comment by SoftVision-CosminMuntean** [comment]
I was pointed to this bug by the mozregression tool but I don't have the privileges to see it's content. So I can't help you with this.
I've tried to contact bechen on IRC to CC you on the issue. He's the one that merged the fix for that issue but he didn't responded to my messages. IRC bot seen him 6 days ago so he might be not available this days. If you want, I can send him an email or something. Just let me know.

**Comment by ianb** [comment]
This reproduced easily for me.  Example crash: https://crash-stats.mozilla.com/report/index/79eddeda-7899-4e98-9fba-4f9932160921

I'm not sure about what to look at next, but I'll try asking around.

**Comment by SoftVision-CosminMuntean** [comment]
I have found another website  were the browser crashes after pressing the Page Shot icon from the toolbar.

**Comment by SoftVision-CosminMuntean** [comment]
I have retested this issue and is no longer reproducible. Tested on the latest Firefox (49.0.2) release, latest Beta (50.0b10), latest Aurora (51.0a2) and latest Nightly (52.0a1) build. 
Considering this, I have performed a find fix regression. Here are the results:
- First good revision: 905f1bc96fb67f7a827387203cb5cafce70b67bf
- Last bad revision: 09943651fa859ca1c208a0682e34dd32e7a7d8f5
- Pushlog: https://goo.gl/dqqx0x

Looks like the following bug has the changes which introduced the fix:
https://bugzilla.mozilla.org/show_bug.cgi?id=1308862

It seems that bug [1308862](https://bugzilla.mozilla.org/show_bug.cgi?id=1308862) fixed this issue.
